services/rtc/flat-rtc-agora-electron/src/rtc-share-screen.ts

```diff
@@ -0,0 +1,207 @@
+import {
+    FlatRTCShareScreen,
+    FlatRTCShareScreenInfo,
+    FlatRTCShareScreenParams,
+} from "@netless/flat-rtc";
+import type AgoraRtcEngine from "agora-electron-sdk";
+import { SideEffectManager } from "side-effect-manager";
+import { combine, Val } from "value-enhancer";
+import { FlatRTCAgoraElectron, FlatRTCAgoraElectronUIDType } from "./flat-rtc-agora-electron";
+
+const rect = { x: 0, y: 0, width: 0, height: 0 };
+
+const videoSourceParams = {
+    width: 0,
+    height: 0,
+    bitrate: 0,
+    frameRate: 15,
+    captureMouseCursor: true,
+    windowFocus: false,
+    excludeWindowList: [],
+    excludeWindowCount: 0,
+};
+
+export type RTCShareScreenParams = FlatRTCShareScreenParams<FlatRTCAgoraElectronUIDType>;
+
+export interface RTCShareScreenAvatarConfig {
+    rtc: FlatRTCAgoraElectron;
+    element?: HTMLElement | null;
+}
+
+export class RTCShareScreen extends FlatRTCShareScreen {
+    private readonly _rtc: FlatRTCAgoraElectron;
+    private readonly _sideEffect = new SideEffectManager();
+
+    private readonly _params$ = new Val<RTCShareScreenParams | null>(null);
+    private readonly _enabled$ = new Val(false);
+
+    private readonly _active$ = new Val(false);
+    private readonly _el$: Val<HTMLElement | null>;
+
+    private readonly _screenInfo$ = new Val<FlatRTCShareScreenInfo | null>(null);
+
+    public constructor(config: RTCShareScreenAvatarConfig) {
+        super();
+
+        this._rtc = config.rtc;
+        this._el$ = new Val(config.element ?? null);
+
+        this._sideEffect.addDisposer(
+            this._active$.subscribe(active => {
+                if (this._el$.value && this._params$.value) {
+                    const { uid } = this._params$.value;
+                    try {
+                        if (active) {
+                            this.client.muteRemoteVideoStream(uid, false);
+                            this.client.setupRemoteVideo(uid, this._el$.value);
+                            this.client.setupViewContentMode(uid, 1, undefined);
```

Comment:
miss comment: https://github.com/netless-io/flat/pull/1508/files#diff-9f20ca5978f513a4bb59b0984343a1bbee1b64beb83dd0d8dc212d5dd39ff86bL20-L23

Reply:
There's no comment before, I have no idea what these codes mean :(

Reply:
```
        // this is a bug in agora SDK, when the `desktop` screen sharing is done,
        // and then the `web` side does the screen sharing,
        // the `desktop` will have a black screen.
        // this is because the SDK has `mute` the remote screen sharing stream
```

Reply:
ok.